File: apps/trading_dashboard/frontend/src/types/experiment.ts
export interface ExperimentSummary {
  run_id: string;
  name: string;
  run_type: string;
  path: string;
  created_at_utc: string | null;
  asset: string | null;
  timeframe: string | null;
  config_hash_sha256: string | null;
  processed_dataset_id: string | null;
  processed_dataset_path: string | null;
  has_trades: boolean;
  has_equity: boolean;
  metrics: Record<string, unknown>;
}

export interface ExperimentDetail {
  run_id: string;
  name: string;
  run_type: string;
  path: string;
  metadata: Record<string, unknown>;
  config: Record<string, unknown>;
  metrics: Record<string, unknown>;
  artifacts: Array<Record<string, unknown>>;
  available_predictions: string[];
  available_trades: string[];
  available_equity: string | null;
  processed_dataset_id: string | null;
  processed_dataset_path: string | null;
}

File: apps/trading_dashboard/backend/app/schemas/experiment.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class ExperimentSummary(BaseModel):
    run_id: str
    name: str
    run_type: str = "experiment"
    path: str
    created_at_utc: str | None = None
    asset: str | None = None
    timeframe: str | None = None
    config_hash_sha256: str | None = None
    processed_dataset_id: str | None = None
    processed_dataset_path: str | None = None
    has_trades: bool
    has_equity: bool
    metrics: dict[str, Any]


class ExperimentDetail(BaseModel):
    run_id: str
    name: str
    run_type: str = "experiment"
    path: str
    metadata: dict[str, Any]
    config: dict[str, Any]
    metrics: dict[str, Any]
    artifacts: list[dict[str, Any]]
    available_predictions: list[str]
    available_trades: list[str]
    available_equity: str | None = None
    processed_dataset_id: str | None = None
    processed_dataset_path: str | None = None


class TradeRecord(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    entry_time: str | None = None
    exit_time: str | None = None
    side: str
    entry_price: float | None = None
    exit_price: float | None = None
    pnl: float | None = None
    return_value: float | None = Field(default=None, alias="return")
    size: float | None = None
    exit_reason: str | None = None
